src/main/java/MealDemo/service/MealPlanningServiceImpl.java

```diff
@@ -82,4 +80,84 @@ public void updateMeal(int mealId, Meal meal) {
         System.out.println("meal has been updated. Meal details \n" + existingMeal.toString());
     }
 
-}
\ No newline at end of file
+
+    @Override
+    public Map<String, List<Meal>> mealsByCategories() {
```

Comment:
I would move this method to `MealRepository` it has to do with data retrieval and can potentially be reused by other service classes in the future.